TidepoolOnboarding/Support/AttributedString.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  AttributedString.swift
+//  TidepoolOnboarding
+//
+//  Created by Darin Krauss on 5/10/21.
+//  Copyright © 2021 Tidepool Project. All rights reserved.
+//
+
+import Foundation
+
+struct AttributedString {
+    enum Attribute {
+        case bold
+        case italic
+    }
+
+    struct Fragment {
+        let string: String
+        let attributes: Set<Attribute>?
+
+        init(_ string: String, attributes: Set<Attribute>? = nil) {
+            self.string = string
+            self.attributes = attributes?.isEmpty == false ? attributes : nil
+        }
+    }
+
+    let fragments: [Fragment]
+
+    init(_ string: String) {
+        self.fragments = [Fragment(string)]
+    }
+
+    init(attributed string: String) {
+        self.fragments = Parser().parse(string) ?? [Fragment(string)]
+    }
+
+    var string: String { fragments.reduce(String(), { $0 + $1.string }) }
+
+    private class Parser: NSObject, XMLParserDelegate {
+        private var fragments: [Fragment] = []
+        private var attributes: [Attribute] = []
+        private var error: Error?
+
+        func parse(_ string: String) -> [Fragment]? {
+            guard let data = "<normal>\(string)</normal>".data(using: .utf8) else {
+                return nil
+            }
+
+            let parser = XMLParser(data: data)
+            parser.delegate = self
+            guard parser.parse() else {
+                return nil
+            }
+
+            return !fragments.isEmpty ? fragments : nil
+        }
+
+        func parser(_ parser: XMLParser, didStartElement elementName: String, namespaceURI: String?, qualifiedName qName: String?, attributes attributeDict: [String: String] = [:]) {
+            switch elementName {
+            case "b":
+                attributes.append(.bold)
+            case "em":
+                attributes.append(.italic)
+            default:
+                break
+            }
+        }
+
+        func parser(_ parser: XMLParser, didEndElement elementName: String, namespaceURI: String?, qualifiedName qName: String?) {
+            switch elementName {
+            case "b", "em":
+                _ = attributes.popLast()
```

Comment:
That is invalid XML so it will fail to parse and will default to using the string as-is. This is as designed. (That is, if you screw up the XML you don't get any attributes, just the original string.)

Reply:
Hm.  Well, that's an implementation detail IMHO...  it is valid in HTML, and, IMHO, unexpected nonetheless.  I realize you don't want to make a general-purpose parser, but this is a limitation that folks familiar with HTML will not realize.

Reply:
Where does it indicate it is HTML? I make no claim that this supports HTML text.

Reply:
I understand, @darinkrauss , I'm just saying that these tags, which _look_ like HTML, might be misconstrued by someone who expects it to behave like HTML, which I do not think is an unreasonable assumption.  I'm just asking that you document that as a limitation.